src/wrap/mod.rs

```diff
@@ -0,0 +1,113 @@
+//! Wrap
+//!
+//! Re-wrap doc comments for prettier styling in code
+
+use anyhow::Result;
+use serde::{Deserialize, Serialize};
+
+use crate::checker::Checker;
+use crate::Documentation;
+use crate::{Detector, Suggestion, SuggestionSet};
+use crate::Span;
+use crate::LineColumn;
+
+/// Parameters for wrapping doc comments
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct WrapConfig {
+    /// Hard limit for absolute length of lines
+    max_line_length: usize,
+}
+
+impl Default for WrapConfig {
+    fn default() -> Self {
+        WrapConfig {
+            max_line_length: 70,
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct Wrapper {}
+
+impl Checker for Wrapper {
+    type Config = WrapConfig;
+    fn check<'a, 's>(docu: &'a Documentation, config: &Self::Config) -> Result<SuggestionSet<'s>>
+    where
+        'a: 's,
+    {
+        let wrapper = textwrap::Wrapper::new(config.max_line_length)
+            .subsequent_indent(" ")
+            .initial_indent(" ");
+        let suggestions = docu.iter().try_fold::<SuggestionSet, _, Result<_>>(
+            SuggestionSet::new(),
+            |mut acc, (origin, chunks)| {
+                for chunk in chunks {
+                    let mut new_lines = chunk
+                        .as_str()
+                        .split("\n\n")
+                        .collect::<Vec<&str>>()
+                        .iter()
+                        .map(|s| s.replace("\n", "") )
+                        .fold::<Vec<String>, _>(Vec::new(), |mut acc, comment| {
+                            let mut new_comment = wrapper
+                                .wrap_iter(comment.trim())
+                                .map(|b| b.into_owned()).collect();
+                            acc.append(&mut new_comment);
+                            acc.push("".into());
+                            acc
+                        });
+                    // remove last newline
+                    let _ = new_lines.pop();
+                    // @todo find proper span and range
+                    let range = 0..chunk.as_str().len();
+                    let span = Span { start: LineColumn { line: 0, column: 0}, end: LineColumn { line: 1, column: 5}};
```

Comment:
Take a look @ `fn CheckableChunk::find_spans(..)` (that's how other checkers are implemented) or https://github.com/drahnr/cargo-spellcheck/blob/master/src/span.rs#L115-L129

Reply:
The thing here is that I need a span which starts at the smallest line and column and ends at the last line and column. So basically I need to find the smallest and biggest entry in the indexmap. I implemented that manually, but do you see an easier way there?

Reply:
https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.min_by

works with iterators to extract the minimum, the same exists for max, you can implement both of them using scan https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.scan

Reply:
I was not aware of that, that explains the name :facepalm: